v4/source/saml.yaml

```diff
@@ -252,3 +252,51 @@
           $ref: "#/components/responses/Forbidden"
         "501":
           $ref: "#/components/responses/NotImplemented"
+  /saml/reset_auth_data:
+    post:
+      tags:
+        - SAML
+      summary: Reset AuthData to Email
+      description: >
+        Reset the AuthData field of SAML users to their email. This is meant to be
+        used when the "id" attribute is set to an empty value ("") from a previously
+        non-empty value.
+
+        ##### Permissions
+
+        Must have `manage_system` permission.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                include_deleted:
+                  type: boolean
+                  default: false
+                  description: Whether to include deleted users.
+                dry_run:
+                  type: boolean
+                  default: false
+                  description: If set to true, the number of users who would be affected is returned.
```

Comment:
Sorry, I missed this before and gave some bad advice. Dry-run should default to "true", but we don't want to default a boolean to true, so we should turn this flag into something like..."execute" and default it to false. That way they have to set a flag in order to manipulate the database.

Reply:
As discussed offline, we will be keeping the API parameters the same as this would be more intuitive.